SANSILL: Input information metadata

darominski opened the pull request:
**Description of work.**

This PR implements adding more metadata regarding inputs and corrections used to the output workspace and then ASCII exports' headers. Both workspace name and filenames building input/correction workspaces have been added to the metadata of reduced samples, with general methods for extracting information placed in `SANSILLCommon`. Stitch factors used in `SANSILLAutoProcess` are now extracted and added to the stitched workspace as `stitch_scale_factors` double list entry, for `SANSILLMultiProcess` these were already being extracted, so only placing stitch factors in the metadata was required. A couple of typos/inconsistencies have been spotted and fixed.

The choice which metadata should be placed in the header of ASCII exports is specified per instrument in relevant IPFs. IPFs of instruments that requested the update have been updated.

**To test:**

1. Code review.
2. Run some v1 and v2 examples:

- Using D11 system tests:

```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *

beams = '2866,2867+2868,2878'
containers = '2888+2971,2884+2960,2880+2949'
container_tr = '2870+2954'
beam_tr = '2867+2868,2867+2868,2867+2868'
samples = ['2889,2885,2881',
           '2887,2883,2879',
           '3187,3177,3167']
# sample_tr = '2871,2869,3172'
sample_tr = ['2871,2871,2871', '2869', '3172']
thick = [0.1, 0.2, 0.2]

# reduce samples
for i in range(len(samples)):
    SANSILLAutoProcess(
        SampleRuns=samples[i],
        BeamRuns=beams,
        ContainerRuns=containers,
        MaskFiles='mask1.nxs,mask2.nxs,mask3.nxs',
        SensitivityMaps='sens-lamp.nxs',
        SampleTransmissionRuns=sample_tr[i],
        ContainerTransmissionRuns=container_tr,
        TransmissionBeamRuns=beam_tr,
        SampleThickness=thick[i],
        OutputWorkspace='iq_s' + str(i + 1),
        StitchReferenceIndex=1
    )
```

Make sure that:
   - beam_ws, beam_numors, mask_ws, transmission_ws, sample.transmission, container_ws, container_numors, are present in the sample logs
   - for the stitched workspace, the stitch scale factors are present in the sample logs

Likewise with the v2 example:

```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

cads = ['17285', '17284', '17082']
trbeam = '17097'
emptybeams = ['17112', '17097', '17098']
cans = ['17109', '17095', '17099']
ctr = '17096'
masks = ['mask3m', 'mask_8m', 'mask_34m']
flat_fields = ['flat_field_3m', 'flat_field_8m', 'flat_field_8m']
edge = 'edge_mask'

sampleTr = ['17162', '17166', '17166']
samplesD1 = ['17251', '17177', '17257']
samplesD2 = ['17192', '17194', '17200']
samplesD3 = ['17180+17184', '17182+17186', '17208+17212']

SANSILLMultiProcess(SampleRunsD1=','.join(samplesD1),
                            SampleRunsD2=','.join(samplesD2),
                            SampleRunsD3=','.join(samplesD3),
                            DarkCurrentRuns=','.join(cads),
                            EmptyBeamRuns=','.join(emptybeams),
                            EmptyContainerRuns=','.join(cans),
                            SampleTrRunsW1=','.join(sampleTr),
                            ContainerTrRuns=ctr,
                            TrEmptyBeamRuns=trbeam,
                            TrDarkCurrentRuns=cads[1],
                            DefaultMask=edge,
                            BeamStopMasks=','.join(masks),
                            FlatFields=','.join(flat_fields),
                            NormaliseBy='Time',
                            CalculateResolution='DirectBeam',
                            BinningFactor=2,
                            OutputWorkspace='out')
```

Fixes #33764 . <!-- and fix #xxxx or close #xxxx xor resolves #xxxx -->
<!-- alternative
*There is no associated issue.*
-->

Release note updated.

<!-- Ensure the base of this PR is correct (e.g. release-next or master)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

darominski committed:
[SANSILL] Removed redundant get_run_number, use one from common

darominski committed:
[SANSILL] Corrected variable name typo

darominski committed:
[SANSILL] Added methods to add correction info to metadata

darominski committed:
[SANSILL] Added methods to add input numors to metadata

in SANS ILL reduction workflow v1.

darominski committed:
[SANSILL] Sample transmission value added to metadata

in SANS ILL workflow v1.

darominski committed:
[SANSILL] Added methods to add input numors to metadata

in SANS ILL workflow v2.

darominski committed:
[SANSILL] Sample transmission value added to metadata

in SANS ILL workflow v2.

darominski committed:
[SANSILL] Added input metadata to ascii export list for D11, D22

darominski committed:
[SANSILL] Added stich scale factors to stitch output metadata

darominski committed:
[SANSILL] Release note for increased metadata on inputs

darominski committed:
[SANSILL] Added input metadata to ascii export list for D33

darominski committed:
[SANSILL] Release note typo correction

sf1919 commented:
Please can the release note for this be moved to the relevant 6.5 folder? Many thanks.

darominski committed:
Merge branch 'main' into 33764_sansill_more_ascii_exports

darominski committed:
Moved release notes to v6.5

MathieuTillet commented on Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction2.py:
```diff
@@ -229,6 +230,45 @@ def PyInit(self):
+            'mask_ws',
+            common.return_numors_from_path(self.getPropertyValue('MaskWorkspace')),
+            True)
+
```
Could this function and the similar one in SANSILLReduction be moved to SANSILLCommon instead of being duplicated? Since it now also contains `add_correction_numors`, that seems more coherent and clearer

MathieuTillet replied:
Thanks for the change!

darominski committed:
[SANSILL] Moved logic of setting additional metadata to common class.

Now, the methods in SANSILLReduction v1 and v2 set only the relevant fields in a dictionary
which is subsequently passed to a function in SANSILLCommon loops over the dictionary and
sets the desired parameters.

MathieuTillet reviewed (approved):
Works as advertised, no complaints with the code, that's good to go
